app/channel.go

```diff
@@ -2337,6 +2337,24 @@ func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string) (*model.
 		return nil, err
 	}
 
+	if post.RootId != "" {
```

Comment:
This should only happen if the AutoFollow is enabled and in the future if the setting for the collapsed threads is off.